cpp/ql/test/query-tests/Critical/MissingCheckScanf/test.cpp

```diff
@@ -36,8 +36,30 @@ int main()
 	{
 		int i = 0;
 
+		scanf("%d", &i); // BAD. Design choice: already initialized variables shouldn't make a difference.
+		use(i);
+	}
+
+	{
+		int i;
+		use(i);
+
+		if (scanf("%d", &i) == 1) // GOOD: only care about uses after scanf call
+		{
+			use(i);
+		}
+	}
+
+	{
+		int i; // Reused variable
+
 		scanf("%d", &i); // BAD
 		use(i);
+
+		if (scanf("%d", &i) == 1) // GOOD
+		{
+			use(i);
+		}
```

Comment:
Should we have a further copy of the first part (the `// BAD` bit) after the `// GOOD` part?  In a way `i` would have been "initialized" by the `// GOOD` part, but we would perhaps still consider the third part BAD for the same reason as line 39?

Reply:
Seems reasonable.